[JHBuild] Update GStreamer modules after GStreamer update to 1.18.4

dpino opened the pull request:
#### 4d45c6d995e03a6282138f3382f0712190da8a20
<pre>
[JHBuild] Update GStreamer modules after GStreamer update to 1.18.4
<a href="https://bugs.webkit.org/show_bug.cgi?id=255509">https://bugs.webkit.org/show_bug.cgi?id=255509</a>

Reviewed by Philippe Normand.

On 21th April, Ubuntu 20.04 will go out of support and GStreamer required
minimum version will be upgraded to 1.18.4 (Debian 11). Systems that
cannot satisfy this depedency can still use JHBuild to provision
required GStreamer dependencies. This changeset updates JHBuild&apos;s
gstreamer.modules to provision version 1.18.4.

Besides that, this changeset also cleans up gstreamer.modules file and
removes unreferenced downstream patches no longer needed.

* Tools/gstreamer/jhbuild.modules:
* Tools/gstreamer/patches/base-0001-playbin-Handle-error-message-with-redirection-indica.patch: Removed.
* Tools/gstreamer/patches/good-0001-qtdemux-Specify-REDIRECT-information-in-error-messag.patch: Removed.
* Tools/gstreamer/patches/gst-plugins-bad-0001-h264parse-Post-a-WARNING-when-data-is-broken.patch: Removed.
* Tools/gstreamer/patches/gst-plugins-bad-0006-mssdemux-parse-protection-data.patch: Removed.
* Tools/gstreamer/patches/gst-plugins-base-Fix-include-xdg-shell-client-protocol.diff: Added.
* Tools/gstreamer/patches/gst-plugins-good-0002-Check-if-an-upstream-demuxer-provided-a-default-kid.patch: Removed.
* Tools/gstreamer/patches/libav-0001-gstavviddec-Limit-default-number-of-decoder-threads.patch: Removed.
* Tools/gstreamer/patches/openh264-configure.patch: Removed.
* Tools/gstreamer/patches/orc-satisfy-fno-common-under-gcc10.patch: Removed.
* Tools/jhbuild/jhbuild-minimal.modules:
* Tools/jhbuild/jhbuild-wrapper:

Canonical link: <a href="https://commits.webkit.org/263216@main">https://commits.webkit.org/263216@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/76389cebf2d96026ce6f29427975c6ebcb5edbfa

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/3373 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/3433 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/3553 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/4804 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/3705 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/3352 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/3502 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/3461 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/2940 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/3414 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/3692 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/3044 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/4622 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/39/builds/1200 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/3005 "Passed tests") | [❌ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/2932 "3 flakes 169 failures") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/2978 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/3063 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/4364 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/3446 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/17/builds/2770 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/3008 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/3004 "Passed tests") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/3015 "Built successfully") | | | 
| [❌ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/452 "Failed to push commit to Webkit repository") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/3284 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

philn commented on Source/cmake/GStreamerChecks.cmake:
```diff
@@ -36,7 +36,7 @@ if (ENABLE_VIDEO OR ENABLE_WEB_AUDIO)
           endif ()
 
-          find_package(GStreamer 1.16.2 REQUIRED COMPONENTS ${GSTREAMER_COMPONENTS})
+          find_package(GStreamer 1.18.4 REQUIRED COMPONENTS ${GSTREAMER_COMPONENTS})
```
This is supposed to be done in the other patch?

dpino replied:
Yes, I can get rid of this part. I put it here to test the PR.

dpino committed:
[JHBuild] Update GStreamer modules after GStreamer update to 1.18.4
https://bugs.webkit.org/show_bug.cgi?id=255509

Reviewed by Philippe Normand.

On 21th April, Ubuntu 20.04 will go out of support and GStreamer required
minimum version will be upgraded to 1.18.4 (Debian 11). Systems that
cannot satisfy this depedency can still use JHBuild to provision
required GStreamer dependencies. This changeset updates JHBuild's
gstreamer.modules to provision version 1.18.4.

Besides that, this changeset also cleans up gstreamer.modules file and
removes unreferenced downstream patches no longer needed.

* Tools/gstreamer/jhbuild.modules:
* Tools/gstreamer/patches/base-0001-playbin-Handle-error-message-with-redirection-indica.patch: Removed.
* Tools/gstreamer/patches/good-0001-qtdemux-Specify-REDIRECT-information-in-error-messag.patch: Removed.
* Tools/gstreamer/patches/gst-plugins-bad-0001-h264parse-Post-a-WARNING-when-data-is-broken.patch: Removed.
* Tools/gstreamer/patches/gst-plugins-bad-0006-mssdemux-parse-protection-data.patch: Removed.
* Tools/gstreamer/patches/gst-plugins-base-Fix-include-xdg-shell-client-protocol.diff: Added.
* Tools/gstreamer/patches/gst-plugins-good-0002-Check-if-an-upstream-demuxer-provided-a-default-kid.patch: Removed.
* Tools/gstreamer/patches/libav-0001-gstavviddec-Limit-default-number-of-decoder-threads.patch: Removed.
* Tools/gstreamer/patches/openh264-configure.patch: Removed.
* Tools/gstreamer/patches/orc-satisfy-fno-common-under-gcc10.patch: Removed.
* Tools/jhbuild/jhbuild-minimal.modules:
* Tools/jhbuild/jhbuild-wrapper:

Canonical link: https://commits.webkit.org/263216@main
